pkg/frontend/transport/handler.go

```diff
@@ -49,13 +49,15 @@ var (
 
 // Config for a Handler.
 type HandlerConfig struct {
-	LogQueriesLongerThan time.Duration `yaml:"log_queries_longer_than"`
-	MaxBodySize          int64         `yaml:"max_body_size" category:"advanced"`
-	QueryStatsEnabled    bool          `yaml:"query_stats_enabled" category:"advanced"`
+	LogQueriesLongerThan   time.Duration `yaml:"log_queries_longer_than"`
+	LogQueryRequestHeaders string        `yaml:"log_query_request_headers" category:"advanced"`
```

Comment:
There's a `flagext.StringSliceCSV` type in dskit that can be used for this.

Reply:
https://github.com/grafana/mimir/pull/5030/commits/c2a427febd95a38c1c4aee38284e0e064bb78ef3